scripts/src/admin/wrapper.html

```diff
@@ -0,0 +1,365 @@
+			let toastElemnt;
+			let previewElemnt;
+			let previewBtnElemnt;
+			let publishBtnElemnt;
+			var overlayJsonEditor;
+
+			document.addEventListener("DOMContentLoaded", function() {
+				overlayElement = document.querySelector(".overlay");
+				overlayTextElement = document.querySelector(".overlay .text");
+				previewBtnElemnt = document.querySelector(".overlay .preview-btn");
+				previewElemnt = document.querySelector(".overlay .preview iframe");
+				publishBtnElemnt = document.querySelector(".overlay .publish-btn");
+
+				toastElemnt = document.querySelector("#toast");
+				previewBtnElemnt.addEventListener("click", refreshPreviewBtnHandler)
+				publishBtnElemnt.addEventListener("click", publishBtnHandler)
+			});
+
+			function toast(msg) {
+				toastElemnt.innerText = msg;
+				toastElemnt.className = "show";
+				setTimeout(function(){ toastElemnt.className = toastElemnt.className.replace("show", ""); }, 3000);
+			}
+			function submitData(url, data) {
+				axios.post(url, data)
+					.then(res => toast("ok"))
+					.catch(err => alert("error: " + JSON.stringify(err)));
+			}
+			function overlayOn(text, offerId) {
+				var jsonEditorOptions = {
+					mode: 'form',
+					modes: ['view', 'form', 'tree', 'code']
+
+				};
+				overlayElement.style.display = "block";
+				overlayElement.dataset.offerId = offerId;
+
+				overlayJsonEditor = new JSONEditor(overlayTextElement, jsonEditorOptions, JSON.parse(unescape(text), null,  2));
+				overlayJsonEditor.expandAll();
+			}
+			function overlayOff() {
+				overlayElement.style.display = "none";
+				overlayTextElement.textContent = "";
+				previewElemnt.src = "";
+			}
+
+			function stringifyJson(json){
+				// This isn't really needed because the JSON editor escapes and protects but just to be on the safe side
+				try {
+					jsonData = JSON.stringify(json);
+					return jsonData;
+				} catch {
+					toast("Invaid JSON");
+					return false;
+				}
+			}
+
+			function editAppConfigs(json, appId) {
+				console.log(json)
+				var jsonEditorOptions = {
+					mode: 'form',
+					modes: ['view', 'form', 'tree', 'code']
+
+				};
+				overlayElement.style.display = "block";
+				overlayElement.dataset.appId = appId;
+
+				overlayJsonEditor = new JSONEditor(overlayTextElement, jsonEditorOptions, JSON.parse(unescape(json)));
+				overlayJsonEditor.expandAll();
+			}
+
+
+			function publishBtnHandler(){
+				const json = stringifyJson(overlayJsonEditor.get());
+				if (!json){
+					return;
+				}
+				var data = {
+					content: json,
+				};
+				submitData("/offers/" + overlayElement.dataset.offerId, data);
+			}
+
+			function refreshPreviewBtnHandler (){
+				refreshPreview(overlayJsonEditor.get());
+			}
+			function refreshPreview(json){
+				let jsonData = stringifyJson(json);
+				if (!jsonData){
+					return;
+				}
+				previewElemnt.src = "${ webview }?cacheBuster=${ now }&jsonData=" + encodeURIComponent(jsonData);
+			}
+
+			function openEditor(jsonArea, appId) {

```

Comment:
I would split this into a function that handles the UI and one that just opens the editor (and accepts just text and element) and is called by the first function.